benchmarks/graphics_pipeline/main.cpp

```diff
@@ -743,6 +847,74 @@ void ProjApp::Setup()
     }
 }
 
+void ProjApp::RepeatGeometryNonPositionVertexData(Geometry& dstGeom, const Geometry& srcGeom, size_t repeatCount)
+{
+    size_t nVertexBufferCount = srcGeom.GetVertexBufferCount();
+    PPX_ASSERT_MSG(nVertexBufferCount == dstGeom.GetVertexBufferCount(), "Mismatched source and destination vertex data format");
+
+    bool copyFirstBuffer;
```

Comment:
Perhaps we could simplify this a little? 
```
PPX_ASSERT(nVertexBufferCount > 0, "Geometry cannot have 0 vertex buffers");
// If there is one interleaved (1 vb), repeat position data as well
// For position planar (2 vb) & planar (1+ vb), repeat only non-position vertex data, starting from buffer 1
size_t firstBufferToCopy = (nVertexBufferCount == 1) ? 0 : 1;

for (size_t vertexBufferIndex = firstBufferToCopy; vertexBufferIndex < nVertexBufferCount; vertexBufferIndex++) {
```

Reply:
Changed, thanks